src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -94,17 +96,26 @@ public class MeterProviderSdk
                 {
                     if (meterSourcesToSubscribe.ContainsKey(instrument.Meter.Name))
                     {
-                        var index = Interlocked.Increment(ref this.metricIndex);
-                        if (index >= MaxMetrics)
+                        // lookup views for this instrument (and meter,type etc).
+                        var viewNames = this.viewConfig[instrument.Name];
+                        var countMetricsToBeCreated = viewNames.Length;
+                        var metrics = new Metric[countMetricsToBeCreated];
+                        for (int i = 0; i < countMetricsToBeCreated; i++)
                         {
-                            // Log that all measurements are dropped from this instrument.
-                        }
-                        else
-                        {
-                            var metric = new Metric(instrument, temporality);
-                            this.metrics[index] = metric;
-                            listener.EnableMeasurementEvents(instrument, metric);
+                            var index = Interlocked.Increment(ref this.metricIndex);
+                            if (index >= MaxMetrics)
```

Comment:
we can decide if we have enough room for *all* streams, else drop it in entirety.